components/brave_vpn/browser/brave_vpn_service.cc

```diff
@@ -805,7 +807,9 @@ void BraveVpnService::OnPrepareCredentialsPresentation(
 
   // Early return when it's already expired.
   if (time < base::Time::Now()) {
-    SetPurchasedState(env, PurchasedState::INVALID);
```

Comment:
Just curious, we reach here when we got valid credential summary - https://github.com/brave/brave-core/blob/master/components/brave_vpn/browser/brave_vpn_service.cc#L730

When we got valid credential summary, can we get expired  credential cookie? @spylogsster @bsclifton

Reply:
It is not possible to reproduce manually, becasue we have few checks for expired date before in our and in Rust code, but this check may protect us from errors in our code, it is covered by tests.